File: barbeariaEntrega2OFC/barbearia-app/src/app/services/feed/post-feed.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PostFeed, ComentarioFeed } from '../../model/feed/post-feed.model';

@Injectable({
  providedIn: 'root'
})
export class PostFeedService {

  private apiUrl = 'http://localhost:8086/api/feed';

  constructor(private http: HttpClient) { }

  save(post: PostFeed): Observable<PostFeed> {
    return this.http.post<PostFeed>(this.apiUrl, post);
  }

  findAll(): Observable<PostFeed[]> {
    return this.http.get<PostFeed[]>(this.apiUrl);
  }

  findById(id: number): Observable<PostFeed> {
    return this.http.get<PostFeed>(`${this.apiUrl}/${id}`);
  }

  update(id: number, post: PostFeed): Observable<PostFeed> {
    return this.http.put<PostFeed>(`${this.apiUrl}/${id}`, post);
  }

  deleteById(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }



  findByAutor(idCliente: number): Observable<PostFeed[]> {
    return this.http.get<PostFeed[]>(`${this.apiUrl}/autor/${idCliente}`);
  }

  pesquisarPorConteudo(termo: string): Observable<PostFeed[]> {
    return this.http.get<PostFeed[]>(`${this.apiUrl}/pesquisar?termo=${termo}`);
  }

  curtirPost(id: number): Observable<PostFeed> {
    return this.http.post<PostFeed>(`${this.apiUrl}/${id}/curtir`, {});
  }

}

File: barbeariaEntrega2OFC/Barbearia4Periodo-master/Barbearia4Periodo-master/src/main/java/app/service/PostFeedService.java
package app.service;

import app.entity.PostFeed;
import app.entity.ComentarioFeed;
import app.repository.PostFeedRepository;
import app.repository.ComentarioFeedRepository;
import app.repository.ClienteRepository;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

@Service
public class PostFeedService {

    private final PostFeedRepository postFeedRepository;
    private final ComentarioFeedRepository comentarioFeedRepository;
    private final ClienteRepository clienteRepository;

    @Autowired
    public PostFeedService(
            PostFeedRepository postFeedRepository,
            ComentarioFeedRepository comentarioFeedRepository,
            ClienteRepository clienteRepository) {
        this.postFeedRepository = postFeedRepository;
        this.comentarioFeedRepository = comentarioFeedRepository;
        this.clienteRepository = clienteRepository;
    }

    public PostFeed save(PostFeed post) {
        if (post.getAutor() != null && post.getAutor().getId_cliente() != null) {
            if (post.getAutor().getId_cliente() > 0) {
                post.setAutor(clienteRepository.findById(post.getAutor().getId_cliente())
                        .orElseThrow(() -> new EntityNotFoundException("Cliente não encontrado")));
            } else {
                // Para Master (id_cliente = 0), usar autorNome e autor = null
                post.setAutorNome("Route 48");
                post.setAutor(null);
            }
        } else {
            throw new IllegalArgumentException("Autor é obrigatório");
        }
        return postFeedRepository.save(post);
    }

    public List<PostFeed> findAll() {
        List<PostFeed> posts = postFeedRepository.findAll();
        posts.sort((a, b) -> {
            if (a.getDataPost() == null && b.getDataPost() == null) return 0;
            if (a.getDataPost() == null) return 1;
            if (b.getDataPost() == null) return -1;
            return b.getDataPost().compareTo(a.getDataPost());
        });
        return posts;
    }

    public Optional<PostFeed> findById(Long id) {
        return postFeedRepository.findById(id);
    }

    public PostFeed update(Long id, PostFeed postDetails) {
        PostFeed post = findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Post não encontrado com id: " + id));

        post.setConteudo(postDetails.getConteudo());
        return postFeedRepository.save(post);
    }

    public void deleteById(Long id) {
        postFeedRepository.deleteById(id);
    }



    public List<PostFeed> findByAutor(Long idCliente) {
        return postFeedRepository.findByAutorId(idCliente);
    }

    public List<PostFeed> pesquisarPorConteudo(String termo) {
        return postFeedRepository.findByConteudoContainingIgnoreCase(termo);
    }


}

